release/release.go

```diff
@@ -37,41 +37,33 @@ type OS struct {
 	VersionID string `json:"version-id,omitempty"`
 }
 
+var (
+	versionSignaturePath = "/proc/version_signature"
+	apparmorSysPath      = "/sys/kernel/security/apparmor/"
+)
+
 // ForceDevMode returns true if the distribution doesn't implement required
 // security features for confinement and devmode is forced.
-func (os *OS) ForceDevMode() bool {
-	supported := []struct {
-		ID        string
-		VersionID string
-	}{
-		{"neon", ""},
-		{"ubuntu", ""},
-		{"ubuntu-core", ""},
-		// NOTE: elementary is "Elementary OS" but normalization
-		// applies
-		{"elementary", "0.4"},
-		// NOTE: mint uses "LinuxMint" (mixed capitalization)
-		// but this is normalized by readOSRelease.
-		{"linuxmint", "18.1"},
-		{"galliumos", "2.0"},
-		{"peppermint", "7.0"},
-		// NOTE: zorin is "Zorin OS" but normalization applies
-		{"zorin", "12"},
+func (o *OS) ForceDevMode() bool {
+	// Check if kernel signature contains "Ubuntu", currently only
+	// the Ubuntu kernels have all the required apparmor patches
+	// (but those are getting upstreamed so at some point we need
+	// to make this check smater)
+	versionSig, err := ioutil.ReadFile(versionSignaturePath)
+	if err != nil {
+		return true
+	}
+	if !strings.HasPrefix(string(versionSig), "Ubuntu ") {
+		return true
 	}
 
-	for _, s := range supported {
-		if os.ID == s.ID {
-			res, err := strutil.VersionCompare(os.VersionID, s.VersionID)
-			if err == nil && res >= 0 {
-				return false
-			}
-		}
+	// Also ensure appamor is enabled (cannot use osutil.FileExists() here
+	// because of cyclic imports)
+	if _, err := os.Stat(apparmorSysPath); err != nil {
+		return true
 	}
```

Comment:
There are actually quite a few other considerations besides /sys/kernel/security/apparmor/. See /lib/apparmor/profile-load for details.